src/axom/slic/core/Logger.cpp

```diff
@@ -78,15 +78,7 @@ Logger::~Logger()
 }
 
 //------------------------------------------------------------------------------
-void Logger::abortIfEnabled(message::Level level)
-{
-  if((m_abortOnError && (level == message::Error)) ||
-     (m_abortOnWarning && (level == message::Warning)))
-  {
-    this->flushStreams();
-    m_abortFunction();
-  }
-}
+void Logger::abort() { m_abortFunction(); }
```

Comment:
This change has problems for downstream users.  Since this was collective and is no longer collective.

I am going to have to rethink how we handle this in Serac.  Though we haven't had a problem currently with any calling of error in non-collective ways. To my knowledge we have been good about it so far.

We flush SLIC before terminating it:

```
void exitGracefully(bool error)
{
  if (axom::slic::isInitialized()) {
    serac::logger::flush();
    serac::logger::finalize();
  }

  int mpi_initialized = 0;
  MPI_Initialized(&mpi_initialized);
  int mpi_finalized = 0;
  MPI_Finalized(&mpi_finalized);
  if (mpi_initialized && !mpi_finalized) {
    MPI_Finalize();
  }
  profiling::finalize();

  accelerator::terminateDevice();

  error ? exit(EXIT_FAILURE) : exit(EXIT_SUCCESS);
}